pennylane_sf/tf.py

```diff
@@ -0,0 +1,275 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Strawberry Fields TF backend for PennyLane.
+"""
+from collections import OrderedDict
+from collections.abc import Sequence  # pylint: disable=no-name-in-module
+import uuid
+
+import numpy as np
+import tensorflow as tf
+
+import strawberryfields as sf
+from strawberryfields.backends.tfbackend.states import FockStateTF
+
+# import state preparations
+from strawberryfields.ops import (
+    # Catstate,

```

Comment:
No, this is something we overlooked when we updated SF to only allow real parameters to all gates; the `Catstate` gate still accepts a complex argument! So unfortunately I couldn't get it to work with the TF2 backend (@antalszava)

Reply:
Hmm, I'd imagine that `Catstate` was skipped because it didn't necessarily require a change in Strawberry Fields, right? :thinking: (That is no casting was needed there as calculations were done using complex values).

There seems to be a difference between the signatures of `Catstate` in PennyLane and SF: [PL turned to use real arguments due to parameter domain checks](https://github.com/PennyLaneAI/pennylane/pull/148) whereas SF remained with using complex arguments.

Think there are two alternatives:
1. Making up for the different signatures in this plugin, similarly to how is being done in [`simulator.py`](https://github.com/PennyLaneAI/pennylane-sf/blob/c152793b04c1a90f55e29440054828b872b99a6e/pennylane_sf/simulator.py#L103). The Fock device can currently handle `CatState` this way.

This solution includes adding logic to `StrawberryFieldsTF.apply` (worked locally for `expval`, for `probs` seems to take a bit more tweaking with the subsequent reshaping):
```python
    if operation == "CatState":
        par = (par[0] * np.exp(par[1] * 1j), par[2])
```
2. [Changing the signature in Strawberry Fields](https://github.com/XanaduAI/strawberryfields/pull/441).

Reply:
Update: after looking into this, it seems that only (2) will work